region/maridia/inner-pink/Halfie Climb Room.json

```diff
@@ -383,6 +383,189 @@
       "name": "Base",
       "requires": []
     },
+    {
+      "link": [1, 2],
+      "name": "Come in Shinecharging, Leave Shinecharged (Gravity)",
+      "entranceCondition": {
+        "comeInShinecharging": {
+          "length": 1,
+          "openEnd": 1
+        }
+      },
+      "requires": [
+        "Gravity",
+        "canShinechargeMovementComplex"
+      ],
+      "exitCondition": {
+        "leaveShinecharged": {
+          "framesRemaining": 80
+        }
+      },
+      "unlocksDoors": [
+        {
+          "types": ["super"],
+          "requires": []
+        },
+        {
+          "types": ["missiles", "powerbomb"],
+          "requires": [
+            "never"
+          ]
+        }
+      ]
+    },
+    {
+      "link": [1, 2],
+      "name": "Come in Shinecharging, Leave Shinecharged (Suitless)",
+      "entranceCondition": {
+        "comeInShinecharging": {
+          "length": 0,
+          "openEnd": 1
+        }
+      },
+      "requires": [
+        "canShinechargeMovementComplex"
+      ],
+      "exitCondition": {
+        "leaveShinecharged": {
+          "framesRemaining": 20
+        }
+      },
+      "unlocksDoors": [
+        {
+          "types": ["super"],
+          "requires": []
+        },
+        {
+          "types": ["missiles", "powerbomb"],
+          "requires": [
+            "never"
+          ]
+        }
+      ]
+    },
+    {
+      "link": [1, 2],
+      "name": "Come in Shinecharged, Leave With Spark (Suitless, Top Position)",
+      "entranceCondition": {
+        "comeInShinecharged": {
+          "framesRequired": 130
+        }
+      },
+      "requires": [
+        "canShinechargeMovementComplex",
```

Comment:
Something to keep in mind for comparison is that waiting for a shot during spark wind-up is expected in Very Hard with hero shots in many rooms. I think we were generally only bumping those up to Expert (via "canShinechargeMovementTricky") if the hero shot was far enough from the door that it required a precisely timed spark in order to avoid the shot despawning. I tend to think that waiting for a shot during wind-up is generally fine in Very Hard as long as there's not some other difficult aspect to it that would bump it up.